actions/admin/upgrades/upgrade_comments_access.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * Convert comment annotations to entities
+ * 
+ * Run for 2 seconds per request as set by $batch_run_time_in_secs. This includes
+ * the engine loading time.
+ */
+// from engine/start.php
+global $START_MICROTIME;
+$batch_run_time_in_secs = 2;
+
+// if upgrade has run correctly, mark it done
+if (get_input('upgrade_completed')) {
+	// set the upgrade as completed
+	$factory = new ElggUpgrade();
+	$upgrade = $factory->getUpgradeFromPath('admin/upgrades/commentaccess');
+	if ($upgrade instanceof ElggUpgrade) {
+		$upgrade->setCompleted();
+	}
+
+	return true;
+}
+
+// Offset is the total amount of errors so far. We skip these
+// comments to prevent them from possibly repeating the same error.
+$offset = get_input('offset', 0);
+$limit = 50;
+
+$access_status = access_get_show_hidden_status();
+access_show_hidden_entities(true);
+
+// don't want any event or plugin hook handlers from plugins to run
+$original_events = _elgg_services()->events;
+$original_hooks = _elgg_services()->hooks;
+_elgg_services()->events = new Elgg\EventsService();
```

Comment:
I was just reusing the patterns from the 1.9 comments upgrade here, but direct queries would be faster.  I'm kind of torn about whether we should be letting the events fire on the upgrade.  I think the only reason to not allow them is performance, but that's not likely to be too much of an issue.  Potentially plugins preventing the update, but if they're doing that then this is pointless anyway as the event handler to synchronize them will fail consistently as well.  I guess we can't be responsible for what plugins might do with such an open system.